java/java.source/src/org/netbeans/modules/java/JavaNode.java

```diff
@@ -429,6 +385,147 @@ public String getValue() {
         }
     }
 
+    private static final class RunFileJDKProperty extends PropertySupport.ReadWrite<String> {
+
+        private final PropertyEditorSupport jdkPicker = new JDKPicker();
+        private final DataObject dObj;
+
+        public RunFileJDKProperty(DataObject dObj) {
+            super("runFileJDK", String.class, // NOI18N
+                  "JDK",
+                  "JDK used to run the file.");
+            this.dObj = dObj;
+        }
+
+        @Override
+        public String getValue() {
+            return dObj.getPrimaryFile().getAttribute(FILE_JDK) instanceof String args ? args : "";
+        }
+
+        @Override
+        public void setValue(String text) {
+            try {
+                dObj.getPrimaryFile().setAttribute(FILE_JDK, text);
+            } catch (IOException ex) {
+                LOG.log(Level.WARNING, "Java File does not exist : {0}", dObj.getPrimaryFile().getName()); //NOI18N
+            }
+        }
+
+        @Override
+        public PropertyEditor getPropertyEditor() {
+            return jdkPicker;
+        }
+
+        private static final class JDKPicker extends PropertyEditorSupport {
```

Comment:
couldn't figure out how inline editors work, so I implemented it the generic way

Reply:
You probably don't need the inline editor support or custom editor for this - just return the `String[]` from `getTags()`.

Reply:
no way, this worked!

Reply:
thanks a lot, this looks much better. Updated the screenshot.

I have to say the API for this is not very intuitive. Now it returns a custom editor, but `supportsCustomEditor` is left at false. This will now display the inline editor without the `...` button.